tools/pylib/_boutpp_build/backend.py

```diff
@@ -0,0 +1,216 @@
+
+
+def size(fn):
+    return os.path.getsize(fn)
+
+
+def gettag():
+    thisos = list(packaging.tags.platform_tags())[-1]
+    tag = "-".join(str(next(packaging.tags.sys_tags())).split("-")[:2] + [thisos])
+    return tag
+
+
+def build_wheel(wheel_directory, config_settings=None, metadata_directory=None):
+    print(config_settings, metadata_directory)
+    opts = ""
+    if config_settings is not None:
+        for k, v in config_settings.items():
+            if k == "sdist":
+                continue
+            if k == "useLocalVersion":
+                global useLocalVersion
+                useLocalVersion = False
+                continue
+            if v:
+                opts += f" {k}={v}"
+            else:
+                opts += f" {k}=ON"
+    tag = gettag()
+    whlname = f"boutpp-{getversion()}-{tag}.whl"
+    trueprefix = f"{os.getcwd()}/_wheel_install/"
+    prefix = f"{trueprefix}/boutpp/"
+    run(
+        "cmake -S . -B _wheel_build/ -DBOUT_ENABLE_PYTHON=ON "
+        + f" -DCMAKE_INSTALL_PREFIX={prefix} -DCMAKE_INSTALL_LIBDIR={prefix} -DCMAKE_INSTALL_PYTHON_SITEARCH={trueprefix}"
+        + opts
+    )
+    run(f"cmake --build  _wheel_build/ -j {os.cpu_count()}")
+    run("cmake --install _wheel_build/")
+    distinfo = f"_wheel_install"
+    prepare_metadata_for_build_wheel("_wheel_install", record=True)
+
+    # Do not add --symlink as python's does not extract that as symlinks
+    run(f"cd {trueprefix} ; zip  {wheel_directory}/{whlname} . -rq")
+    # cmd = f"git archive HEAD -o {wheel_directory}/{whlname}"
+    # run(cmd)
+    return whlname
+
+
+def build_sdist(sdist_directory, config_settings=None):
+    print(config_settings, sdist_directory)
+    enable_gz = False
+    enable_xz = True
+    if config_settings is not None:
+        for k, v in config_settings.items():
+            if k == "sdist":
+                if v == "onlygz":
+                    enable_gz = True
+                    enable_xz = False
+                elif v == "both":
+                    enable_gz = True
+                else:
+                    raise ValueError(f"unknown option {v} for {k}")
+            if k == "useLocalVersion":
+                global useLocalVersion
+                useLocalVersion = False
+    prefix = f"boutpp-{getversion()}"
+    name = f"{prefix}.tar"
+    run(f"git archive HEAD --prefix {prefix}/ -o {sdist_directory}/{name}")
+    _, tmp = tempfile.mkstemp(suffix=".tar")
+    for ext in "fmt", "mpark.variant":
+        run(
+            f"git archive --remote=externalpackages/{ext} HEAD --prefix  {prefix}/externalpackages/{ext}/ --format=tar > {tmp}"
+        )
+        run(f"tar -Af {sdist_directory}/{name} {tmp}")
+        run(f"rm {tmp}")
+
+    # _, tmpd = tempfile.mkdtemp()
+    with open(tmp, "w") as f:
+        f.write(
+            f"""Metadata-Version: 2.1
+Name: boutpp
+Version: {getversion()}
+License-File: COPYING
+"""
```

Comment:
No need to change, but you might like to know about [`textwrap.dedent`](https://docs.python.org/3/library/textwrap.html#textwrap.dedent) :)